File: Algorithm_232/DFS.cpp
#include<bits/stdc++.h>
using namespace std;
const int N = 200;int adj[N][N];
int visited[N]={0};

void dfs(int u,int n){
    if(visited[u]) return;
    visited[u]=1;
    cout<<u<<" ";
    for(int i=1; i<=n; i++){
        if(adj[u][i]==1 and !visited[i]){
            dfs(i,n);
        }
    }

}

int main(){
    int n,m; cin>>n>>m;
    for(int i=0; i<m; i++){
        int u,v; cin>>u>>v;
        adj[u][v]=1; adj[v][u]=1;
    }
    cout<<"Enter source node :";
    int src; cin>>src;
    dfs(src,n);
}

File: AI-CSE-321/Lab-Task-4/DFS.py
graph = {'A': [('B',5), ('C', 7)],
         'B': [('D',7),('E', 3)],
         'C': [('F',4)],
         'D': [],
         'E': [],
         'F': []
         }
def dfs(src, dest):
    visited=[]
    stack = []
    stack.append((src, 0))
    while stack:
        node , cost = stack.pop()
        if node not in visited:
            visited.append(node)
            print(node, end=" ")
            if node==dest:
                print("Destination Found")
                return
            for child, weight in reversed(graph[node]):
                if child not in visited:
                    stack.append((child, weight+cost))
        


    print("Destination Not Found")

dfs('A', 'F')

